### PR TITLE
test(staker): add unit tests to staker

### PR DESCRIPTION
**Pull request by jinoosss**
## Descriptions

This PR introduces comprehensive unit test coverage for the staker module, adding 4,425 lines of test code across 12 test files. The tests cover all major functionalities of the staker contract including minting, staking, reward calculations, and pool management.

### Test Files Added

**Core Functionality Tests**
- `mint_stake_test.gno`: Tests for minting and staking positions with various scenarios including GNOT handling, parameter validation, fee tiers, tick ranges, and slippage protection
- `calculate_pool_position_reward_test.gno`: Tests for calculating pool position rewards

**Reward Calculation Tests**
- `reward_calculation_pool_test.gno`: Tests for pool-level reward calculations
- `reward_calculation_incentives_test.gno`: Tests for incentive-based reward calculations
- `reward_calculation_pool_tier_test.gno`: Tests for tier-based reward calculations
- `reward_calculation_tick_test.gno`: Tests for tick-level reward calculations

**Pool Management Tests**
- `manage_pool_tier_and_warmup_test.gno`: Tests for pool tier management and warmup period functionality

**External Integration Tests**
- `external_incentive_test.gno`: Tests for external incentive mechanisms
- `external_token_list_test.gno`: Tests for external token list management
- `external_deposit_fee_test.gno`: Tests for external deposit fee handling

**Protocol Operations Tests**
- `protocol_fee_unstaking_test.gno`: Tests for protocol fee collection during unstaking
- `assert_test.gno`: Tests for validation and assertion functions used throughout the staker module

### Test Coverage

The test suite covers:
- Parameter validation and boundary conditions
- GNOT and GRC20 token handling
- Multiple fee tiers (100, 500, 3000, 10000)
- Tick range validation and tick spacing
- Slippage protection mechanisms
- Deadline validation
- Referrer parameter handling
- Reward calculation algorithms across different scenarios
- Pool tier management and warmup periods
- External incentive integrations
- Protocol fee mechanisms
- Error handling and panic scenarios

### Testing Approach

All tests follow a table-driven testing pattern with comprehensive test cases covering:
- Valid scenarios with expected successful outcomes
- Invalid inputs with expected panic messages
- Edge cases and boundary conditions
- Multiple token pair combinations
- Various fee tier configurations

### CODE REVIEW

File: contract/r/gnoswap/staker/v1/reward_calculation_pool_test.gno [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pool := sr.NewPool(tt.poolPath, tt.currentTime)
+			poolResolver := NewPoolResolver(pool)
+
+			tick := poolResolver.CurrentTick(tt.currentTime)
+
+			// Tick should be a valid int32
+			_ = tick
+		})
```
Comment (notJoon) [inline contract/r/gnoswap/staker/v1/reward_calculation_pool_test.gno]: The return value of `CurrentTick` has not been validated explicitly.
